src/wms/src/main/java/org/geoserver/wms/featureinfo/FreeMarkerTemplateManager.java

```diff
@@ -66,11 +85,43 @@ public TemplateModel wrap(Object object) throws TemplateModelException {
                                     "geoJSON",
                                     getStaticModel(
                                             "org.geoserver.wms.featureinfo.GeoJSONTemplateManager"));
+                            addConfiguredStatics(map);
```

Comment:
This call is made every time a wrapping happens, meaning:

- The indications provided in system variables/env vars/context variables are processed over and over
- All the warning message are repeated for each request

Since the configuration is static by nature (should not change at runtime), all this should be processed once only in a static initializer that builds a map with the necessary configuration. This map is then going to be merged into the SimpleHash every time.

I can see how that would make testing harder, I guess you can add a protected static method "reset(String rulesSpec)" that changes the contents of that map, which the test will use.

Reply:
Yes, you are right, I will adjust it.